src/GoTrueClient.ts

```diff
@@ -673,7 +674,27 @@ export default class GoTrueClient {
       if (!refresh_token) {
         throw new Error('No current session.')
       }
+      // Check if refresh already in progress
+      const tokenRefreshLock =
+        isBrowser() && (await this.localStorage?.getItem(`${STORAGE_KEY}-tokenRefreshLock`))
+      if (tokenRefreshLock === 'IN_PROGRESS') {
+        // Release lock after 5s in case original lock setter has dropped off.
+        if (isBrowser())
+          setTimeout(
+            () => {
+              this.localStorage?.removeItem(`${STORAGE_KEY}-tokenRefreshLock`)
+              this._recoverAndRefresh()
+            },
+            5 * 1000 // remove lock after 5s
```

Comment:
would it help if we randomise this `setTimeout` duration too? in the event there are multiple tabs waiting for the lock, there could be multiple instances of `setTimeout` invoked at the same time?

maybe it's better if the original lock setter calls `setTimeout` to remove the lock after 5 seconds after it acquires the lock [here](https://github.com/supabase/gotrue-js/pull/274/files#diff-3522461172efd6058d6b8da62fc2d30d8b524d2b64894ea2c67218c52f7fdff5R693)

Reply:
On the lock with a timer... What happens if the tab that set the lock gets closed right after that?
I'll admit I'm not up to speed on the need for multiple tabs and what the user can do with these tabs, but they can certainly close one at any point.

Reply:
ok i just tested out the following and closed the tab before the timeout expired:
```
setTimeout(function(){
  console.log("Hello from setTimeout")
}, 5000);
```
seems like once you close the tab, the timeout gets cleared so yeah it needs to be called by another tab